### ISSUE TITLE
Add note about compatibility with c-style for loops to Known Issues

### ISSUE THREAD

**Issue by bourquep** [realm/SwiftLint#530]
Repro:

``` swift
for var i = 0;
    i < 10;
    i++
{
}
```
- The `trailing_semicolon` rule incorrectly flags those semicolons as being superfluous.
- When using the `autocorrect` option, those semicolons are removed.

**Comment by jpsim** [comment]
Seeing as how the c-style for loops are on their way out, deprecated in Swift 2.2, I'm not sure how critical this is.

But if someone (maybe @bourquep?) takes the time to fix this, I'll gladly accept a patch!

**Comment by scottrhoyt** [comment]
Hmm. Yeah, you are right. This type of loop is deprecated in Swift 2.2 and will be removed in Swift 3.0. However, 3.0 is months away, and we wouldn't want to break a bunch of people's code before that. On the other hand, that will add significant complexity to that rule, so it's not a trivial call to make either. Swift 2.2 should be out in a month or two, I suppose we could recommend not using this rule if still using the old c-style loops.

**Comment by scottrhoyt** [comment]
Because I don't want to see someone spend a chunk of time implementing something this temporary (not to mention affecting the performance of all SwiftLint users), my recommendation would be that this is a good point to add a "Known Issues" section to the README. We can note that this rule only works correctly if using the soon-to-be-correct style of for loops. In that way, I guess we'd be staying true to promoting good Swift style! :smile:

**Comment by bourquep** [comment]
>In that way, I guess we'd be staying true to promoting good Swift style!

I agree! I had not realized that this type of loop would soon become deprecated. In my case, I have only 3 such loops in my earlier Swift source files and I can easily refactor them.

**Comment by scottrhoyt** [comment]
Great. Going to rename this issue to track adding this to Known Issues.

**Comment by jpsim** [comment]
I'm going to close this since Swift 2.x is just losing importance every day. Even if we worked to improve this today, by the time we released it Swift 2.x would be even less important.
